aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -213,17 +330,52 @@ module aptos_framework::coin {
         account_addr: address,
         amount: u64,
         dst_coin: &mut AggregatableCoin<CoinType>,
-    ) acquires CoinStore {
+    ) acquires CoinStore, CoinConversionMap, CoinInfo, PairedCoinType {
         // Skip collecting if amount is zero.
         if (amount == 0) {
             return
         };
 
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        let coin = extract(&mut coin_store.coin, amount);
+        let coin_balance = coin_balance<CoinType>(account_addr);
+        let (coin_amount_to_collect, fa_amount_to_collect) = if (coin_balance >= amount) {(amount, 0)} else {(coin_balance, amount - coin_balance)};
+        let coin = zero();
+        if (coin_amount_to_collect > 0) {
+            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
+            merge(&mut coin, extract(&mut coin_store.coin, coin_amount_to_collect));
+        };
+        if (fa_amount_to_collect > 0) {
+            let store = primary_fungible_store::primary_store(account_addr, paired_metadata<CoinType>());
+            let fa = fungible_asset::withdraw_internal(object::object_address(&store), fa_amount_to_collect);
+            merge(&mut coin, fungible_asset_to_coin<CoinType>(fa));
+        };
         merge_aggregatable_coin(dst_coin, coin);
     }
 
+    fun maybe_convert_to_fungible_store<CoinType>(
```

Comment:
How about a `public entry fun` version that simply accepts a signer and passes the address of it to this function, so that users can trigger conversion of a coin store in case they want to interact with new protocols that require users to have a primary fungible asset store?